spec/requests/waste_carriers_engine/business_type_forms_spec.rb

```diff
@@ -49,17 +39,340 @@ module WasteCarriersEngine
+            end
+          end
+
+          # TODISCUSS: Consider shared scenario?
+          context "when the transient registration does not belongs to the logged in user" do
+            let(:transient_registration) do
+              create(
+                :transient_registration,
+                :has_required_data
+              )
+            end
+
+            it "redirects to account permission page" do
+              get new_business_type_form_path(transient_registration[:reg_identifier])
+
+              expect(response).to redirect_to(page_path(:permission))
+            end
+          end
+        end
+      end
+
+      # TODO: Consider sharerd scenario for this only
+      context "when no user is signed in" do
+        it "redirects to the login page" do
+          get new_business_type_form_path(:reg_identifier)
+
+          expect(response).to redirect_to(new_user_session_path)
+        end
+      end
+    end
+
+    ### POST FORM - START
+    # include_examples "POST form",
+    #                  "business_type_form",
+    #                  valid_params: { business_type: "limitedCompany" },
+    #                  invalid_params: { business_type: "foo" },
+    #                  test_attribute: :business_type
+    describe "POST business_type_form_path" do
+      context "when a valid user is signed in" do
+        let(:user) { create(:user) }
+
+        before(:each) do
+          sign_in(user)
+        end
+
+        context "when no renewal is in progress" do
+          let(:registration) do
+            create(:registration,
+                   :has_required_data,
+                   :expires_soon,
+                   account_email: user.email)
+          end
+
+          let(:params) do
+            { reg_identifier: registration.reg_identifier }
+          end
+
+          it "redirects to the renewal_start_form and does not create a transient_registration" do
+            post_form_with_params(:business_type_form, params)
+
+            expect(response).to redirect_to(new_renewal_start_form_path(registration[:reg_identifier]))
+
+            matching_transient_regs = WasteCarriersEngine::TransientRegistration.where(reg_identifier: registration.reg_identifier)
+            expect(matching_transient_regs.any?).to be_falsey
+          end
+
+          # TODISCUSS
+          context "when the linked registration does not belongs to the logged in user" do
+            let(:registration) do
+              create(
+                :registration,
+                :has_required_data,
+                :expires_soon
+              )
+            end
+
+            it "redirects to account permission page" do
+              post_form_with_params(:business_type_form, reg_identifier: registration[:reg_identifier])
+
+              expect(response).to redirect_to(page_path(:permission))
             end
           end
         end
+
+        context "when a renewal is in progress" do
```

Comment:
Unit tests on conroller for testing authorization, integration tests on cancancan abilities to check that the authorize code gives the expected output